src/utils.ts

```diff
@@ -184,9 +186,32 @@ export function errorHandling(errorDetails: any, label?: string, moreInfo?: stri
             break;
         }
         default: {
-            window.showErrorMessage(moreInfo);
+            window.showErrorMessage(moreInfo + " " +  errorDetails);
             break;
         }
     }
     return;
 }
+
+/*************************************************************************************************************
+ * Refresh Profile and Session
+ * @param {sessNode} IZoweTreeNode
+ * @param {profile} IProfileLoaded
+ *************************************************************************************************************/
+export function refreshTree(sessNode: IZoweTreeNode) {
+    const allProf = Profiles.getInstance().getProfiles();
+    for (const profNode of allProf) {
+        if (sessNode.getProfileName() === profNode.name) {
+            sessNode.getProfile().profile = profNode.profile;
+            const SessionProfile = profNode.profile as ISession;
+            if (sessNode.getSession().ISession !== SessionProfile) {
+                sessNode.getSession().ISession.user = SessionProfile.user;
+                sessNode.getSession().ISession.password = SessionProfile.password;
+                sessNode.getSession().ISession.base64EncodedAuth = SessionProfile.base64EncodedAuth;
+                sessNode.getSession().ISession.hostname = SessionProfile.hostname;
+                sessNode.getSession().ISession.port = SessionProfile.port;
+                sessNode.getSession().ISession.rejectUnauthorized = SessionProfile.rejectUnauthorized;
```

Comment:
Maybe we should have a setter for this?
I know it won't have much of an performance impact, but theoretically this could reduce the amount of session calls to one :smile:

@jellypuno, @Colin-Stone, any thoughts?

No change requested!

Reply:
I would vote for a setter